### ISSUE TITLE
wrong height for iconPanels #mm-subblocker 

### ISSUE THREAD

**Issue by apptaro** [BeSite/jQuery.mmenu#387]
Tested with mmenu 5.3.1. Here is the test case:
http://jsfiddle.net/apptaro/7vjxsb51/

1) open menu
2) scroll down menu for five menu items
3) click to open submenu for menu8
4) confirm that the bottom five items are not blocked by #mm-subblocker

By the way, thank you for this wonderful library.

**Comment by FrDH** [comment]
This is a known issue, I'm not yet sure how to fix this other than with a nasty workaround.

**Comment by apptaro** [comment]
Looking at the code, I understand that the height of mm-subblocker is set by top:0px and bottom0px, so it equals to the height of the panel, not the height of the panel contents.
There is no standard way to fix the position of the mm-subblocker relative to its parent. It's a bad idea to update the position listening to scroll events.
So a better approach would be to set the height of mm-subblocker when it is shown. But the corresponding logic is in the core (jquery.mmenu.oncanvas.js,) thus a nasty workaround.

I'm trying to hack it around by editing openPanel function in jquery.mmenu.oncanvas.js, line 155:
```javascript
if ( $panel.hasClass( _c.opened ) )
{
	$panel.nextAll( '.' + _c.opened )
		.addClass( _c.highest )
		.removeClass( _c.opened )
		.removeClass( _c.subopened )
		.children( '.' + _c.subblocker ).css("height", "");
}
else
{
	$panel.addClass( _c.highest );
	$current.addClass( _c.subopened );
	$current.children( '.' + _c.subblocker ).height($current[0].scrollHeight);
}
```
I'm not finished, but this might work. (But it's hacky)

Another approach would be to hide mm-subblocker as default:
```css
.mm-subblocker { display: none; }
.mm-subopened .mm-subblocker { display: block; }
```
Then set the height of mm-subblocker inside jquery.mmenu.iconpanels.js, line 70:
```javascript
$curp
	.removeClass( _c.hidden )
	.add( $panl )
	.slice( -opts.visible )
	.each(
		function( x )
		{
			$(this).addClass( _c.iconpanel + '-' + x );
			$(this).children( '.' + _c.subblocker ).height($(this)[0].scrollHeight);
		}
	);
```
I think this approach will be a clean solution. What do you think?

**Comment by FrDH** [comment]
You're probably better off using the API to bind a new function to the `openPanel` method. That way you don't need to hack into the core.

This might not be a waterproof solution, but setting the `bottom` of the subblocker to `-10000px` and `overflow-y` of the panel to `hidden` also seems to work OK (at least on FF, Chrome and Safari on a desktop). I would have thought setting `overflow-y: hidden;` would jump the panel back to the top, but it didn't.
Could you have a go at this approach and confirm it works?

**Comment by apptaro** [comment]
I never thought this bug could be fixed by only css. I created a pull request. Thank you very much for your wonderful work.

**Comment by FrDH** [comment]
This fix has been implemented in version 5.3.2
